6.5.group_by.py

```diff
@@ -0,0 +1,11 @@
+def group_by(f, itr):
+    # pass each item in itr to f by using a generator expression
```

Comment:
When you document a function you should also document the parameters it receives.